onnxruntime/core/framework/data_types.cc

```diff
@@ -845,33 +845,33 @@ const NonTensorTypeBase* DataTypeImpl::SequenceTensorTypeFromONNXEnum(int type)
 const SparseTensorTypeBase* DataTypeImpl::SparseTensorTypeFromONNXEnum(int type) {
   switch (type) {
     case TensorProto_DataType_FLOAT:
-      return reinterpret_cast<const SparseTensorTypeBase*>(DataTypeImpl::GetSparseTensorType<float>());
+      return static_cast<const SparseTensorTypeBase*>(DataTypeImpl::GetSparseTensorType<float>());
```

Comment:
Can we just do `DataTypeImpl::GetSparseTensorType<float>()->AsSparseTensorType();` instead ?

Reply:
You are totally right! Tried and replaced by that :)

Reply:
Thanks. I had the same update in #8339.